### PR TITLE
HPCC-31281 Warn if configuration may lead to excessive tracing

### PR DESCRIPTION
**Pull request by richardkchapman**
Some Roxie configuration settings are designed for use only while temporarily diagnosing an issue, and should not be left on permanently due to excessive logging impact.

Generate a warning if such options are set.

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [x] Security
  - [x] Thread-safety
  - [x] Cloud-compatibility
  - [x] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:
Examined helm output manually

### CODE REVIEW

**Review by jakesmith — APPROVED** [review]
@richardkchapman - looks good.

If we warn for all expert settings in roxie configs,
should we also do so for other components?

File: helm/hpcc/templates/_warnings.tpl [hunk 1]
```diff
@@ -73,6 +73,42 @@ Pass in dict with root and warnings
...
+   {{- end -}}
+  {{- end -}}
+ {{- end -}}
+ {{- /* Warn if any expert flags are set */ -}}
+ {{- range $cname, $ctypes := .root.Values -}}
+  {{- range $id, $component := $ctypes -}}
+   {{- if kindIs "map" $component -}}
+    {{- if not $component.disabled -}}
+     {{- range $key, $value := $component.expert -}}
+      {{- if $value -}}
```
Comment (ghalliday) [inline helm/hpcc/templates/_warnings.tpl:102]: Is this check useful?  We can revisit if any export options start defaulting to true.

**Review by ghalliday — APPROVED** [review]
Looks good.  Would be even better if there was a test case in testing/helm.
Please squash.

**Review by mckellyln — COMMENTED** [review]
Looks ok to me, but should this logic be in ccdmain for bare-metal (or both) ?

File: helm/hpcc/templates/_warnings.tpl (hunk 1 above)
Comment (richardkchapman) [reply helm/hpcc/templates/_warnings.tpl:102]: It is useful for developers who leave expert options in place but disabled...

**Comment by richardkchapman** [comment]
@ghalliday Squashed

**Review by ghalliday — APPROVED** [review]

**Pull request merged**